app/src/main/java/org/openedx/app/AppExternalRouter.kt

```diff
@@ -0,0 +1,292 @@
+            }
+
+            "course_handout" -> {
+                externalParams["course_id"]?.let { courseId ->
+                    appRouter.navigateToHandoutsWebView(
+                        fm = fm,
+                        courseId = courseId,
+                        type = HandoutsType.Handouts
+                    )
+                }
+            }
+
+            "course_announcement" -> {
+                externalParams["course_id"]?.let { courseId ->
+                    appRouter.navigateToHandoutsWebView(
+                        fm = fm,
+                        courseId = courseId,
+                        type = HandoutsType.Announcements
+                    )
+                }
+            }
+
+            "course_component" -> {
+                externalParams["course_id"]?.let { courseId ->
+                    externalParams["component_id"]?.let { componentId ->
+                        launch {
+                            try {
+                                courseInteractor.preloadCourseStructure(courseId)
+                                val courseStructure = courseInteractor.getCourseStructureFromCache()
+
+                                courseStructure.blockData
+                                    .find { it.descendants.contains(componentId) }?.let { block ->
+                                        appRouter.navigateToCourseContainer(
+                                            fm = fm,
+                                            courseId = courseId,
+                                            unitId = block.id,
+                                            componentId = componentId,
+                                            mode = CourseViewMode.FULL
+                                        )
+                                    }
+                            } catch (e: Exception) {
+                                e.printStackTrace()
+                            }
+                        }
+                    }
+                }
+            }
+
+            // Program
+            "program" -> {
+                val pathId = externalParams["path_id"]
+                if (pathId == null) {
+                    appRouter.navigateToMain(
+                        fm = fm,
+                        courseId = null,
+                        infoType = null,
+                        selectedTab = HomeTab.PROGRAMS
+                    )
+                } else {
+                    appRouter.navigateToEnrolledProgramInfo(
+                        fm = fm,
+                        pathId = pathId
+                    )
+                }
+            }
+
+            // Discussions
+            "discussion_topic" -> {
+                externalParams["course_id"]?.let { courseId ->
+                    externalParams["topic_id"]?.let { topicId ->
+                        launch {
+                            try {
+                                discussionRepository.getCourseTopics(courseId)
```

Comment:
Could we use interactor here instead of repo?

Reply:
done